### DISCUSSION TITLE
Separate Git Repository for Mozilla SOPS

### DISCUSSION THREAD

**Discussion by robincher** [fluxcd/flux2#2158] [Q&A] [answered]
Hi,

I managed to bootstrap Fluxv2 in my Gitlab repository (using a MonoRepo pattern for various clusters),and its working well. 

Additionally, we also implemented SOPS to managed our config secrets. As such, i am making this Git Repository as the source of truth for all my code and its secrets.

I am just thinking if there is possibility (might have missed the doc/concept), where i can decoupled the encrypted secrets into a separate repository?  In this manner, i can have a better segregation of duty between the platform engineers that concentrate on deploying resources and configs, while the developers can check-in the required secrets for their container in another repository.

**Comment by stefanprodan** [comment, marked as answer]
> I am just thinking if there is possibility (might have missed the doc/concept), where i can decoupled the encrypted secrets into a separate repository?

Yes but decoupling the secrets from the deployments means that you can’t use Kustomize generator to restart the deployments when a secret changes so you’ll need https://github.com/stakater/Reloader

**Reply by robincher** [reply]
Hi @stefanprodan ,

Don't mind me going a bit into the implementation details.

So that mean existing clusters are able to have two "source", one for configs, and another one for secrets.

If that the case,do you mean i gotta run another bootstrap command on the existing cluster to the new "secrets" repository?

```
flux bootstrap gitlab \
  --hostname=gitlab.com
  --token-auth \
  --owner=somewhere
  --repository=secrets-gitops \
  --branch=master \
  --path=clusters/cluster-name
```

**Reply by stefanprodan** [reply]
> do you mean i gotta run another bootstrap command on the existing cluster

No bootstrap must be run only once per cluster, see the get started guide on how to add more repos: https://fluxcd.io/docs/get-started/

**Reply by robincher** [reply]
Thanks, will close this, totally missed out on that podinfo repository part :)
